### PR TITLE
#274 Added script to output makefile-dependencies for Fortran modules.

### LINKED ISSUE #274: Add example for Fortran dependency detection
Show how to use fparser to create dependencies between Fortran files suitable for a Makefile, e.g.:
```
$ ./create_dependencies.py *.x90 *.f90 *.F90
assign_coordinate_field_mod.o: base_mesh_config_mod.o planet_config_mod.o
configuration_mod.o: base_mesh_config_mod.o extrusion_uniform_config_mod.o \
    finite_element_config_mod.o io_utility_mod.o partitioning_config_mod.o \
    perturbation_bell_config_mod.o planet_config_mod.o timestepping_config_mod.o
init_perturbation_kernel_mod.o: perturbation_bell_config_mod.o
prop_perturbation_kernel_mod.o: perturbation_bell_config_mod.o
time_evolution_alg_mod.o: init_perturbation_kernel_mod.o \
    prop_perturbation_kernel_mod.o timestepping_config_mod.o
```
It will be based on the assumption that a module ``xx`` will be in a file ``xx.f90``, ``xx.F90``, or ``xx.x90`` (the latter for PSyclone support).

### PR DESCRIPTION
**Pull request by hiker**
Fixes #274 - adds the example of dependency creator for makefiles.

### CODE REVIEW

File: example/create_dependencies.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# ------------------------------------------------------------------------------
+# Author: Joerg Henrichs, Bureau of Meteorology
+
+'''This file contains an fparser script that parses Fortran files
+and output the dependencies between these files suiteable for a Makefile.
```
Comment (arporter) [inline example/create_dependencies.py:39]: "suitable"

File: example/create_dependencies.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
```
Comment (arporter) [inline example/create_dependencies.py:1]: This won't work on a system that only has 'python3' but I'm not sure what we do about that.
Maybe just add a README (which we should have in the examples directory anyway) and mention this limitation.

**Review by arporter — CHANGES_REQUESTED** [review]
Looks good - thanks Joerg. I think the only thing missing is a (short) README.md to explain what the script does. If you'd be so kind as to add a few lines for the `fparser_f2008.py` script that would be great.

Separately (possibly a new issue), we should really be testing these examples and currently we don't.

File: example/create_dependencies.py (hunk 1 above)
Comment (hiker) [reply example/create_dependencies.py]: Done.

File: example/create_dependencies.py (hunk 2 above)
Comment (hiker) [reply example/create_dependencies.py:1]: Indeed. I've added it to the README as known issue.

**Comment by hiker** [comment]
> Looks good - thanks Joerg. I think the only thing missing is a (short) README.md to explain what the script does. If you'd be so kind as to add a few lines for the `fparser_f2008.py` script that would be great.

Done - hope I got it right. I've also added a known-issues and TODO list - maybe someone gets motivated (and I might do it if I need it 😁  ).

> 
> Separately (possibly a new issue), we should really be testing these examples and currently we don't.

Agreed, I've opened #279 ... actually, I just added a very simple makefile that executes the examples (same way as in PSyclone - i.e. it's executed, but we don't test if the output is as expected). But I did not integrate it into travis, since I assume we are moving to github CI anyway. I'll mention this in #279.

**Review by arporter — APPROVED** [review]
All good now. Thanks for adding the Makefile and README.

**Pull request merged**